pkg/archive/gzip.go

```diff
@@ -47,6 +47,12 @@ type GzipStabilizer struct {
 	Func func(*MutableGzipHeader)
 }
 
+func (g GzipStabilizer) Stabilize(arg any) {
+	if g.Func != nil {
```

Comment:
i'd probably just assume Func is set and have a "loud" outcome (i.e. a crash) if there's a probable misconfiguration rather than a "quiet" one where the entry is skipped.